syntax = "proto2";

package opentxs.proto;
option optimize_for = LITE_RUNTIME;

enum SignatureRole {
    SIGROLE_ERROR = 0;
    SIGROLE_PUBCREDENTIAL = 1;
    SIGROLE_PRIVCREDENTIAL = 2;
    SIGROLE_NYMIDSOURCE = 3;
    SIGROLE_CLAIM = 4;
    SIGROLE_SERVERCONTRACT = 5;
    SIGROLE_UNITDEFINITION = 6;
    SIGROLE_PEERREQUEST = 7;
    SIGROLE_PEERREPLY = 8;
}

enum CredentialType {
    CREDTYPE_ERROR = 0;
    CREDTYPE_LEGACY = 1;
    CREDTYPE_HD = 2;
}

enum CredentialRole {
    CREDROLE_ERROR = 0;
    CREDROLE_MASTERKEY = 1;
    CREDROLE_CHILDKEY = 2;
    CREDROLE_CONTACT = 3;
    CREDROLE_VERIFY = 4;
}

enum AsymmetricKeyType {
    AKEYTYPE_ERROR = 0;
    AKEYTYPE_NULL = 1;
    AKEYTYPE_LEGACY = 2;
    AKEYTYPE_SECP256K1 = 3;
    AKEYTYPE_ED25519 = 4;
}

enum KeyMode {
    KEYMODE_ERROR = 0;
    KEYMODE_NULL = 1;
    KEYMODE_PUBLIC = 2;
    KEYMODE_PRIVATE = 3;
}

enum KeyRole {
    KEYROLE_ERROR = 0;
    KEYROLE_AUTH = 1;
    KEYROLE_ENCRYPT = 2;
    KEYROLE_SIGN = 3;
}

enum SourceType {
    SOURCETYPE_ERROR = 0;
    SOURCETYPE_PUBKEY = 1;
    SOURCETYPE_BIP47 = 2;
}

enum SourceProofType {
    SOURCEPROOFTYPE_ERROR = 0;
    SOURCEPROOFTYPE_SELF_SIGNATURE = 1;
    SOURCEPROOFTYPE_SIGNATURE = 2;
}

enum HashType {
    HASHTYPE_ERROR = 0;
    HASHTYPE_NONE = 1;
    HASHTYPE_BTC160 = 2;
    HASHTYPE_BTC256 = 3;
    HASHTYPE_RIPEMD160 = 4;
    HASHTYPE_SHA256 = 5;
    HASHTYPE_SHA512 = 6;
    HASHTYPE_BLAKE2B160 = 7;
    HASHTYPE_BLAKE2B256 = 8;
    HASHTYPE_BLAKE2B512 = 9;
}

enum CredentialIndexMode {
    CREDINDEX_ERROR = 0;
    CREDINDEX_PRIVATE = 1;
    CREDINDEX_PUBLIC = 2;
}
enum CredentialSetMode {
    CREDSETMODE_ERROR = 0;
    CREDSETMODE_INDEX = 1;
    CREDSETMODE_FULL = 2;
}
